### ISSUE TITLE
[Windows] Msix creation failed for an application with a custom library referred directly

### ISSUE THREAD

**Issue by AlanGusLive** [dotnet/maui#9547]
### Description

When a program use a ".Net MAUI Class Library" the creation of the Msix package for Windows generate the following message : "Improper project configuration: WindowsPackageType is set to None, but PublishAppxPackage is set to true."

Related issues:
https://github.com/dotnet/maui/issues/7268
https://github.com/dotnet/maui/issues/5881

### Steps to Reproduce

Create a MAUI Application
Build the solution in Release Mode
With the main project in visual studio, right click with the mouse to call "Publish..."
Select Sideloading and uncheck "Enable automatic updates"
Create a new certificate and press "Next" button
Create a Publishing profile (configuration=Release|Any CPU, Target Runtime=win10-x64, Self Contained= True, Package Location=bin\Release\net6.0-windows10.0.19041.0\win10-x64\AppPackages\)
Click on the "Create" button to generate the package. **It Works.**

**Now** in the solution add a ".Net MAUI Class Library" and refer to the application.
Build the solution in Release Mode
With the main project in visual studio, right click with the mouse to call "Publish..."
Select Sideloading and uncheck "Enable automatic updates"
Press the "Next" button (the certificate is already selected)
Click on the "Create" button to generate the package. **An ERROR occurs** with the message "Improper project configuration: WindowsPackageType is set to None, but PublishAppxPackage is set to true."

[PublishMsix.zip](https://github.com/dotnet/maui/files/9382437/PublishMsix.zip)

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows SDK 10.0.19041.0

### Did you find any workaround?

None

**Comment by lutzroeder** [comment]
@oliag @mariaghiondea @sbanni

**Comment by mattleibow** [comment]
Duplicate of #9879

Unfortunately, there seems to be a bug in the Windows App SDK: https://github.com/dotnet/maui/issues/9879#issuecomment-1238628696

**Comment by mattleibow** [comment]
Actually, this looks like a parallel issue. The error is from the other issue, but the cause may be a bit different. It appears that some of the properties meant for the app is reaching the class library - most notably `penerateAppxPackageOnBuild` as that tells the build system to generate a MSIX, but that is impossible for the class library.

**Comment by AlanGusLive** [comment]
@mattleibow I tested your [workaround](https://github.com/dotnet/maui/issues/9879#issuecomment-1238628696) and it works. I can now create a MSIX to install the package in Windows with an application which contains a custom library referred directly.
Thanks for the workaround.

For information, I noticed that when you add a new ".NET MAUI Class Library" in a project, the Deploy of the library is automaticaly checked in Debug & Release Mode and need to be unchecked.

![image](https://user-images.githubusercontent.com/15076413/189073513-2cfeb101-6a07-4e31-aaa2-2458b9d854ad.png)

**Comment by mattleibow** [comment]
Looking into this a bit more and I see the `<PublishAppxPackage>true</PublishAppxPackage>` property in the publish profile is giving the specific error for:

> Improper project configuration: WindowsPackageType is set to None, but PublishAppxPackage is set to true.

This seems to be because all those properties in that file get applied to _all_ project in the publish. I am not sure I agree with that, but that is a separate issue.

Once I removed this property, we fall back into the issue we always had:

https://github.com/microsoft/WindowsAppSDK/issues/3337

**Comment by homeyf** [comment]
Verified this issue with Visual Studio Enterprise 17.8.0 Preview 1.0(.NET8). Can repro this issue with sample project.
[PublishMsix.zip](https://github.com/dotnet/maui/files/9382437/PublishMsix.zip)
![image](https://github.com/dotnet/maui/assets/82450723/64cd65f9-cdd1-4f7e-be29-e284b4dc2516)

**Comment by anlocnghg** [comment]
> This is still an issue in `net8.0.3`. Any chances this is fixed soon? I added a simple repro here. [#8304 (comment)](https://github.com/dotnet/maui/issues/8304#issuecomment-1838262839)

Just a note that, I tried removing the reference in the csproj file. "It works" in the sense that the solution compiles, but actually it does not work, because then all the references and using statements (from the razor class library project) do not work in the main/startup project anymore.

**Comment by ArpitViradiyaCrowncon** [comment]
Any update on this?

**Comment by ArpitViradiyaCrowncon** [comment]
Any update on this?

**Comment by MateuszLas421** [comment]
any update ?

**Comment by kubaflo** [comment]
Closing as external. This thread is blocked on behavior outside the normal MAUI backlog (WindowsAppSDK/publish/publish-profile behavior) and has remained open across multiple releases without an in-repo fix path. If upstream behavior changes and there is still a current MAUI-specific gap, please file a fresh issue.

**Comment by kubaflo** [comment]
Reopening — this issue should not have been closed.

It carries `s/verified` (a maintainer or partner reproduced it), `s/triaged` (it already went through triage), `partner/winui`, `partner/cat 😻` (a partner team is tracking it). Those labels mean the report was already confirmed as a real product defect, so it is not eligible for backlog cleanup closure. Closure here requires specific fix evidence (a merged PR that closes it, or a `fixed-in-*` label), and there was none.

Sorry for the noise, and thanks to everyone who flagged it.

> Disclosure: this comment was generated by AI.
